host-macros/src/service.rs

```diff
@@ -129,10 +130,60 @@ impl ServiceBuilder {
 
     /// Construct instructions for adding a characteristic to the service, with static storage.
     fn construct_characteristic_static(&mut self, characteristic: Characteristic) {
-        let name_screaming = format_ident!(
-            "{}",
-            inflector::cases::screamingsnakecase::to_screaming_snake_case(characteristic.name.as_str())
-        );
+        let descriptors: TokenStream2 = characteristic
+            .args
+            .descriptors
+            .iter()
+            .enumerate()
+            .map(|(index, args)| {
+                let name_screaming =
+                    format_ident!("DESC_{}_{index}", to_screaming_snake_case(characteristic.name.as_str()));
+                let properties = set_desc_access_properties(&args);
+                let uuid = args.uuid;
+                // let read_callback = args.on_read.as_ref();
+                // let write_callback = args.on_write.as_ref();
+                let writeable = args.write || args.write_without_response;
+                let capacity = match &args.capacity {
+                    Some(cap) => quote!(#cap),
+                    None => {
+                        if writeable {
+                            panic!("'capacity' must be specified for writeable descriptors");
+                        }
+                        quote!(0u8)
```

Comment:
Shouldn't the capacity be set to the length of the default value (if available) or some predefined size if it's set dynamically? (e.g. 32 bytes or something)

Reply:
That assumes the default value is the longest it'll ever be though?

Reply:
Sorry what I mean is that if the capacity isn't explicitly set by the user we need to make some sort of call on what the capacity should be.
My understanding is that the most common use case for descriptors is that they don't change for the lifetime of the application. If this is accurate, I think the most ergonomic and memory efficient way to handle this is to size the descriptor's memory to the given value. An alternative would be to just have a default capacity, however if we do that it will mean that in almost all cases we are either wasting memory, or causing the user to have to realise their default value is being truncated, and then figure out how to change the capacity.
Either way though, it looks like the code here is setting the capacity to 0, so nothing will be stored?

Reply:
you are correct, I've fixed this now by:

```rust
let default_value = match &args.default_value {
                    Some(val) => quote!(#val), // if set by user
                    None => quote!(""),
                };
                let capacity = match &args.capacity {
                    Some(cap) => quote!(#cap),
                    None => {
                        if writeable {
                            panic!("'capacity' must be specified for writeable descriptors");
                        }
                        quote!(#default_value.len() as u8)
                    },
                };
```